test_util/cluster.py

```diff
@@ -0,0 +1,260 @@
+        Returns:
+            dict[public_IP] = private_IP
+        """
+        ip_detect_script = pkg_resources.resource_filename('gen', 'ip-detect/aws.sh')
+        tunnel.write_to_remote(ip_detect_script, join(self.remote_dir, 'ip-detect.sh'))
+        private_ip = tunnel.remote_cmd(['bash', join(self.remote_dir, 'ip-detect.sh')]).decode('utf-8').strip('\n')
+        assert len(private_ip.split('.')) == 4
+        return private_ip
+
+    def get_cluster_hosts(self, public_ips, setup):
+        ssh_hostports = [i + ':22' for i in public_ips]
+
+        @retry(stop_max_delay=120000)
+        def establish_host_connectivity():
+            """Continually try to recreate the SSH Tunnels to all hosts for 2 minutes
+            """
+            return closing(TunnelCollection(self.ssh_user, self.ssh_key_path, ssh_hostports))
+
+        logging.info('Checking that hosts are accessible')
+        with establish_host_connectivity() as tunnels:
+            private_ip = {}
+            for tunnel in tunnels.tunnels:
+                private_ip[tunnel.host] = self.get_private_address(tunnel)
+                if setup:
+                    # Make the default user priveleged to use docker
+                    tunnel.remote_cmd(['sudo', 'usermod', '-aG', 'docker', self.ssh_user])
+
+        return [Host(public_ip, private_ip[public_ip]) for public_ip in public_ips]
+
+    @contextmanager
+    def host_tunnel(self, host):
+        with closing(SSHTunnel(self.ssh_user, self.ssh_key_path, host.public_ip)) as tunnel:
+            yield tunnel
+
+    def install_dcos(
+        self,
+        installer_url,
+        api=False,
+        setup=True,
+        add_config_path=None,
+        installer_api_offline_mode=True,
+        install_prereqs=True,
+        install_prereqs_only=False,
+    ):
+        if api:
+            installer = test_util.installer_api_test.DcosApiInstaller()
+            installer.offline_mode = installer_api_offline_mode
+        else:
+            installer = test_util.installer_api_test.DcosCliInstaller()
+
+        with self.host_tunnel(self.bootstrap_host) as bootstrap_host_tunnel:
+            logging.info('Setting up installer on bootstrap host')
+
+            installer.setup_remote(
+                tunnel=bootstrap_host_tunnel,
+                installer_path=self.remote_dir + '/dcos_generate_config.sh',
+                download_url=installer_url)
+            if setup:
+                # only do on setup so you can rerun this test against a living installer
+                logging.info('Verifying installer password hashing')
+                test_pass = 'testpassword'
+                hash_passwd = installer.get_hashed_password(test_pass)
+                assert passlib.hash.sha512_crypt.verify(test_pass, hash_passwd), 'Hash does not match password'
+                if api:
+                    installer.start_web_server()
+
+            with open(pkg_resources.resource_filename("gen", "ip-detect/aws.sh")) as ip_detect_fh:
```

Comment:
nit: can be a future refactor, but we're going to want to pass in the ip_detect_script + config dictionary to use when installing DC/OS